src/test/java/io/debezium/connector/spanner/config/validation/ConnectionValidatorTest.java

```diff
@@ -7,51 +7,99 @@
 
-    private static Stream<Arguments> configProvider() {
-        return Stream.of(
-                Arguments.of(
-                        Configuration.from(Map.of(
-                                "gcp.spanner.project.id", "boxwood-weaver-353315",
-                                "gcp.spanner.instance.id", "kafka-connector",
-                                "gcp.spanner.credentials.path", "no_path",
-                                "gcp.spanner.database.id", "kafkaspan")),
-                        true),
-                Arguments.of(
-                        Configuration.from(Map.of()),
-                        false));
+        ConnectionValidator connectionValidator = spy(ConnectionValidator.withContext(validationContext));
+        connectionValidator.validate();
+        Assertions.assertEquals(true, connectionValidator.isSuccess());
     }
 
-    @Disabled
-    @ParameterizedTest
-    @MethodSource("configProvider")
-    void validate(Configuration configuration, boolean isSuccess) {
+    @Test
+    void validateSuccessAgainstEmulator() {
+        Configuration configuration = Configuration.from(Map.of(
+                "gcp.spanner.project.id", "boxwood-weaver-353315",
+                "gcp.spanner.instance.id", "kafka-connector",
+                "gcp.spanner.database.id", "kafkaspan",
+                "gcp.spanner.emulator.host", "http://localhost:9010"));
         Map<String, ConfigValue> configValueMap = Map.of(
-                "gcp.spanner.project.id", new ConfigValue("gcp.spanner.project.id"),
-                "gcp.spanner.instance.id", new ConfigValue("gcp.spanner.instance.id"),
-                "gcp.spanner.database.id", new ConfigValue("gcp.spanner.database.id"),
-                "gcp.spanner.credentials.json", new ConfigValue("gcp.spanner.credentials.json"),
-                "gcp.spanner.credentials.path", new ConfigValue("gcp.spanner.credentials.path"));
+                "gcp.spanner.project.id", new ConfigValue("gcp.spanner.project.id", "boxwood-weaver-353315", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.instance.id", new ConfigValue("gcp.spanner.instance.id", "kafka-connector", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.database.id", new ConfigValue("gcp.spanner.database.id", "kafkaspan", new ArrayList<>(), new ArrayList<>()),
+                "gcp.spanner.emulator.host", new ConfigValue("gcp.spanner.emulator.host", "http://localhost:9010", new ArrayList<>(), new ArrayList<>()));
 
         ConfigurationValidator.ValidationContext validationContext = new ConfigurationValidator.ValidationContext(configuration, configValueMap);
 
         ConnectionValidator connectionValidator = spy(ConnectionValidator.withContext(validationContext));
-        // doNothing().when(connectionValidator).tryConnectByCredentialsJsonPath();
+        connectionValidator.validate();
+        Assertions.assertEquals(false, connectionValidator.isSuccess());
```

Comment:
Shouldn't this be true?

Reply:
Yes you are correct. I have updated the assertion here as well as the logic for credentials check. Now [`canValidate`](https://github.com/ShuranZhang/debezium-connector-spanner/blob/EmulatorTesting/src/main/java/io/debezium/connector/spanner/config/validation/ConnectionValidator.java#L88) will only check for credentials if `isAgainstEmulator` is wrong.